CMakeLists.txt

```diff
@@ -130,32 +129,54 @@ if (WIN32)
         set(VLD_LIBRARY "${VLD_DIR}/lib/Win32/vld.lib")
     endif()
   endif()
-else()
+
+  # set all include directories
+  target_include_directories(${TARGET_NAME} PRIVATE
+    ${VLD_INCLUDE_DIR}
+    ${SDL2_INCLUDE_DIR} 
+    ${SDL2_IMAGE_INCLUDE_DIR} 
+    ${SDL2_TTF_INCLUDE_DIR})
+  
+  # set libraries to link with
+  target_link_libraries(${TARGET_NAME} PUBLIC
+    ${SDL2_LIBRARIES} 
+    ${SDL2_IMAGE_LIBRARIES} 
+    ${SDL2_TTF_LIBRARIES} 
+    ${VLD_LIBRARY}
+  )
+
+elseif(NOT EMSCRIPTEN)
   list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/sdl2)
-endif()
 
+  find_package(SDL2 REQUIRED)
+  find_package(SDL2_image REQUIRED)
+  find_package(SDL2_ttf REQUIRED)
+
+  include_directories(${SDL2_INCLUDE_DIRS})
+  include_directories(${SDL2_image_INCLUDE_DIRS})
+  include_directories(${SDL2_ttf_INCLUDE_DIRS})
+
+  target_link_libraries(${TARGET_NAME} PUBLIC SDL2::Main SDL2::Image SDL2::TTF)
+endif()
 
-# set all include directories
 target_include_directories(${TARGET_NAME} PRIVATE
   ${CMAKE_SOURCE_DIR}/Minigin/
-  ${VLD_INCLUDE_DIR}
-  ${GLM_INCLUDE_DIR}
-  ${SDL2_INCLUDE_DIR} 
-  ${SDL2_IMAGE_INCLUDE_DIR} 
-  ${SDL2_TTF_INCLUDE_DIR})
-
-# set libraries to link with
-target_link_libraries(${TARGET_NAME} 
-  ${SDL2_LIBRARIES} 
-  ${SDL2_IMAGE_LIBRARIES} 
-  ${SDL2_TTF_LIBRARIES} 
-  ${VLD_LIBRARY})
+  ${glm_SOURCE_DIR}
+)
 
+# compile at max warning level + treat warnings as errors
+target_compile_options(${TARGET_NAME} PRIVATE
+  $<$<CXX_COMPILER_ID:MSVC>:/W4 /WX>
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -Wpedantic -Werror>
+)
 
 if(EMSCRIPTEN)
   set(EMSCRIPTEN_COMPILE_FLAGS "${EMSCRIPTEN_LINK_FLAGS} -sUSE_SDL=2 -sUSE_SDL_IMAGE=2 -sUSE_SDL_TTF=2 -g")
   set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS} -sUSE_SDL=2 -sUSE_SDL_IMAGE=2 -sUSE_SDL_TTF=2 -g -sALLOW_MEMORY_GROWTH --preload-file \"${CMAKE_SOURCE_DIR}/Data@/\" -sSDL2_IMAGE_FORMATS=['tga','png']")
 
+  # work around https://github.com/emscripten-core/emscripten/issues/11513
+  set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS} -fno-stack-protector")
```

Comment:
Without this line, the Emscripten build crashes with this log:
```
[100%] Linking CXX executable minigin_web.html
[parse exception: attempted pop from empty stack / beyond block start boundary at 1056407 (at 0:1056407)]
Fatal: error in parsing input
em++: error: '/usr/bin/wasm-emscripten-finalize -g --dyncalls-i64 --pass-arg=legalize-js-interface-exported-helpers --dwarf minigin_web.wasm -o minigin_web.wasm --detect-features' failed (returned 1)
```

Would be good to figure out the root cause (the linked issue says it's fixed...) but for now adding this flag bypasses the crash and allows minigin to run on Firefox.